### DISCUSSION TITLE
¿Could this error be happening because I'm creating the nodes from OnInitializedAsync?

### DISCUSSION THREAD

**Discussion by JuanMaGMZG** [Blazor-Diagrams/Blazor.Diagrams#336] [General]
I am loading some nodes that I have stored in a database. When rendering them, I retrieve these nodes using an asynchronous function.

They render correctly, but upon finishing the rendering, I get this message:

![image](https://github.com/Blazor-Diagrams/Blazor.Diagrams/assets/137310542/9d03b5d4-2be1-4f73-828b-44faa4700eef)

What I'm doing is adding the nodes to a node list, iterating through them, and adding them to the diagram like this:

foreach (DiagramNode node in nodeList)
{
    diagram.Nodes.Add(node);
}

Regards

**Comment by JuanMaGMZG** [comment]
¿the problem?

```
<CascadingValue Value="diagram">
        <DiagramCanvas>
            <Widgets>
                <NavigatorWidget Width="300" Height="200" DefaultStyle="true"></NavigatorWidget>
            </Widgets>
        </DiagramCanvas>
    </CascadingValue>
```

**Reply by zHaytam** [reply]
Hello, what version are you using?

**Reply by JuanMaGMZG** [reply]
2.1.6

**Comment by JuanMaGMZG** [comment]
I have obtained version 3.0.0 and adapted it to the new nomenclatures, and it's working now. Thank you very much.

**Comment by zHaytam** [comment]
Thank you!
